packages/lib/services/profile.ts

```diff
@@ -0,0 +1,81 @@
+import { prisma } from "@formbricks/database";
+import { DatabaseError, ResourceNotFoundError } from "@formbricks/errors";
+import { Prisma } from "@prisma/client";
+import { TProfile, TProfileUpdateOutput } from "@formbricks/types/v1/profile";
+import { cache } from "react";
+
+const responseSelection = {
+  id: true,
+  name: true,
+  email: true,
+};
+
+// function to retrive basic information about a user's profile
+export const getProfile = cache(async (userId: string): Promise<TProfile> => {
+  try {
+    const profile = await prisma.user.findUnique({
+      where: {
+        id: userId,
+      },
+      select: responseSelection,
+    });
+
+    if (!profile) {
+      throw new ResourceNotFoundError("Profile", userId);
+    }
+
+    return profile;
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      throw new DatabaseError("Database operation failed");
+    }
+
+    throw error;
+  }
+});
+
+// function to update a user's profile
+export const updateProfile = async (
+  userId: string,
```

Comment:
We currently have a mix of profile & user. Let's make this all consistent and use `Profile` for a user profile on formbricks. that also helps us solve some of the confusion we have around person & user (the users of our users).
So please rename `userId` to `personId`. Later we will also change the database model of user to profile but we can wait with that :-)